samcli/lib/remote_invoke/remote_invoke_executors.py

```diff
@@ -0,0 +1,224 @@
+"""
+Abstract class definitions and generic implementations for remote invoke
+"""
+import json
+import logging
+from abc import ABC, abstractmethod
+from io import TextIOWrapper
+from pathlib import Path
+from typing import Any, Callable, List, Optional, Union, cast
+
+LOG = logging.getLogger(__name__)
+
+
+class RemoteInvokeExecutionInfo:
+    """
+    Keeps request and response information about remote invoke execution
+
+    payload: payload string given by the customer
+    payload_file: if file is given, this points to its location
+
+    response: response object returned from boto3 action
+    exception: if an exception is thrown, it will be stored here
+    """
+
+    # Request related properties
+    payload: Optional[Union[str, List, dict]]
+    payload_file: Optional[TextIOWrapper]
+
+    # Response related properties
+    response: Optional[Union[dict, str]]
+    exception: Optional[Exception]
+
+    def __init__(self, payload: Optional[Union[str, List, dict]], payload_file: Optional[TextIOWrapper]):
+        self.payload = payload
+        self.payload_file = payload_file
+        self.response = None
+        self.exception = None
+
+    def is_file_provided(self):
+        return self.payload_file
+
+    @property
+    def payload_file_path(self) -> Optional[TextIOWrapper]:
+        return self.payload_file if self.is_file_provided() else None
+
+    def is_succeeded(self):
+        return self.response
```

Comment:
could you please add the return type to the function definition. 
Is it possible to have a failure response?
can you modify the return statement to return boolean if the return type is boolean

Reply:
> Is it possible to have a failure response?
It is possible, and it is kept inside another field called `exception`. That field is only used in tests for now, will be used once we raised follow-up PRs.